java-checks-testkit/src/main/java/org/sonar/java/checks/verifier/JavaCheckVerifier.java

```diff
@@ -217,6 +217,9 @@ private static void scanFile(String filename, JavaFileScanner check, JavaCheckVe
     conf.setJavaVersion(javaCheckVerifier.javaVersion);
     JavaAstScanner.scanSingleFileForTests(new File(filename), visitorsBridge, conf);
     VisitorsBridgeForTests.TestJavaFileScannerContext testJavaFileScannerContext = visitorsBridge.lastCreatedTestContext();
+    if (testJavaFileScannerContext == null) {
+      Fail.fail("Semantic was required but it was not possible to create it. Please checks the logs to find out the reason.");
+    }
     javaCheckVerifier.checkIssues(testJavaFileScannerContext.getIssues(), javaCheckVerifier.providedJavaVersion);
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'testJavaFileScannerContext' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)